here-naksha-lib-ext-manager/src/main/java/com/here/naksha/lib/extmanager/ExtensionCache.java

```diff
@@ -83,15 +85,38 @@ protected void buildExtensionCache(ExtensionConfig extensionConfig) {
       publishIntoCache(result, extensionConfig);
     });
 
-    // Removing existing extension which has been removed from the configuration
-    List<String> extIds =
-        extensionConfig.getExtensions().stream().map(Extension::getId).toList();
+    // Removing existing extension which has been removed from the configuration.
+    List<String> extIds = extensionConfig.getExtensions().stream()
+        .map(extension -> extension.getEnv() + ":" + extension.getId())
+        .toList();
 
     for (String key : loaderCache.keySet()) {
       if (!extIds.contains(key)) {
-        loaderCache.remove(key);
-        PluginCache.removeExtensionCache(key);
-        logger.info("Extension {} removed from cache.", key);
+        KVPair<Extension, ClassLoader> kvPair = loaderCache.get(key);
+        Object instanceObj = kvPair.getValue();
+        Extension extension = kvPair.getKey();
+        final String extensionIdWthEnv = extension.getEnv() + ":" + extension.getId();
+        if (!isNullOrEmpty(extension.getInitClassName())) {
+          if (instanceObj instanceof IExtensionInit initInstance) {
```

Comment:
this condition will never satisfy, meaning instanceObj is expected to be of type ClassLoader and not IExtensionInit. Pls revisit this code. It seems logical error.

Reply:
Corrected it.